rllib/models/preprocessors.py

```diff
@@ -216,6 +216,38 @@ def observation_space(self) -> gym.Space:
         return self._obs_space
 
 
+@PublicAPI
+class MultiBinaryPreprocessor(Preprocessor):
+    """Preprocessor that turns a MultiBinary space into a Box.
+
+    Note: Before RLModules were introduced, RLlib's ModelCatalogV2 would produce
+    ComplexInputNetworks that treat MultiBinary spaces as Boxes. This preprocessor is
+    needed to get rid of the ComplexInputNetworks and use RLModules instead because
+    RLModules lack the logic to handle MultiBinary or other non-Box spaces.
+    """
+
+    @override(Preprocessor)
+    def _init_shape(self, obs_space: gym.Space, options: dict) -> List[int]:
+        return self._obs_space.shape
+
+    @override(Preprocessor)
+    def transform(self, observation: TensorType) -> np.ndarray:
+        # The shape stays the same, but the dtype changes.
+        self.check_shape(observation)
+        return observation.astype(np.float32)
+
+    @override(Preprocessor)
+    def write(self, observation: TensorType, array: np.ndarray, offset: int) -> None:
+        array[offset : offset + self._size] = np.array(observation, copy=False).ravel()
+
+    @property
+    @override(Preprocessor)
+    def observation_space(self) -> gym.Space:
+        obs_space = gym.spaces.Box(0.0, 1.0, self.shape, dtype=np.float32)
+        obs_space.original_space = self._obs_space
+        return obs_space
```

Comment:
Context for this change: 

The new catalog will, by default, support two spaces: 1D and 3D image. Preprocessors cast other spaces we are compatible with into these two. Previously, we would simply cast incoming data to float inside the models. This is an implicit conversion between multi binary and float-box that we don't want to do anymore, so that the observation space of the RLModules fully matches the incoming observations.

This preprocessor has to be excluded from our ModelV2 code paths though in order to not change our old models.